SCons/Variables/__init__.py

```diff
@@ -292,53 +317,72 @@ def Save(self, filename, env) -> None:
                             except TypeError:
                                 defaultVal = option.converter(defaultVal, env)
 
-                        if str(env.subst('${%s}' % option.key)) != str(defaultVal):
-                            fh.write('%s = %s\n' % (option.key, repr(value)))
+                        if str(env.subst(f'${option.key}')) != str(defaultVal):
+                            fh.write(f'{option.key} = {value!r}\n')
                     except KeyError:
                         pass
-        except OSError as x:
-            raise SCons.Errors.UserError('Error writing options to file: %s\n%s' % (filename, x))
+        except OSError as exc:
+            msg = f'Error writing options to file: {filename}\n{exc}'
+            raise SCons.Errors.UserError(msg) from exc
 
-    def GenerateHelpText(self, env, sort=None) -> str:
-        """ Generates the help text for the options.
+    def GenerateHelpText(self, env, sort: Union[bool, Callable] = False) -> str:
+        """Generate the help text for the Variables object.
 
         Args:
             env: an environment that is used to get the current values
-                of the options.
+                of the variables.
             sort: Either a comparison function used for sorting
-                (must take two arguments and return -1, 0 or 1)
+                (must take two arguments and return ``-1``, ``0`` or ``1``)
                 or a boolean to indicate if it should be sorted.
         """
-
+        # TODO the 'sort' argument matched the old way Python's sorted()
+        #   worked, taking a comparison function argument. That has been
+        #   removed so now we have to convert to a key.
         if callable(sort):
             options = sorted(self.options, key=cmp_to_key(lambda x, y: sort(x.key, y.key)))
+
         elif sort is True:
-            options = sorted(self.options, key=lambda x: x.key)
+            options = sorted(self.options)
```

Comment:
Why dropping key=... ?

Reply:
Had to think for a while, since it's been several weeks since I worked up this stuff... It's unneeded because of the addition of an appropriate dunder method in the `Variable` class - see about line 54.

Reply:
Gotcha! Thanks for explaining.